File: maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        # Fixed size sliding window
        # Build initial window of size k and compute its sum
        curr = 0
        for i in range(k):
            curr += nums[i]
        
        # Slide window one element at a time and update the answer
        ans = curr
        n = len(nums)
        for i in range(k, n):
            curr += nums[i] - nums[i - k]
            ans = max(ans, curr)
        
        return ans / k

File: maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double currSum = 0;
        double maxSum = 0;
        
        for(int i = 0; i < k; i++){
            currSum += nums[i];
        }
        
        maxSum = currSum;
        
        for(int i = k; i < nums.length; i++){
            currSum -= nums[i - k];
            currSum += nums[i];
            maxSum = Math.max(maxSum, currSum);
        }
        
        return maxSum / k;
    }
}